### DISCUSSION TITLE
How to delete a selection made by useControl hook by imperatively accessing the methods of the MapboxDraw plugin

### DISCUSSION THREAD

**Discussion by raulrod16124** [uber/react-map-gl#1904] [Q&A]
Hi. I'm using the useControl hook to create custom map controls in a DrawControl component.

https://codesandbox.io/s/react-map-gl-testing-reference-access-xfq8hd?file=/src/MapView.tsx

I want to access imperatively the MapboxDraw plugging methods through a reference on the DrawControl component.
Because I want to delete all draws from an extern button and not through the trash icon.

The Map component already uses a reference to access the method of MapboxMap:

![image](https://user-images.githubusercontent.com/71431792/176216911-9e0e3b9e-f689-45c8-9690-5c26f07530e5.png)
![image](https://user-images.githubusercontent.com/71431792/176216458-fd7c03de-f490-42ec-9458-39a6622d62d7.png)

so, when I want to use a method from the mapRef, works properly:
![image](https://user-images.githubusercontent.com/71431792/176218369-08c249c0-5530-41fb-b5b9-7289961f948e.png)

Then I tried to do the same in the DrawControl component:
![image](https://user-images.githubusercontent.com/71431792/176217901-8dd47ee8-7878-4d56-8fb6-1f95116cc2cf.png)
![image](https://user-images.githubusercontent.com/71431792/176217971-0facaf8b-32f0-407f-97e7-64c21ecaf552.png)

But don´t work:
![image](https://user-images.githubusercontent.com/71431792/176218989-511e44e2-e219-44e9-a681-390cf4f82a83.png)

I have been trying to assign a reference inside the DrawControl component where I use the useControl hook, but I don't know-how

Could you tell me how I can imperatively access the methods of the MapboxDraw plugin? So I can delete all the draws, through the .deleteAll( ) method, without clicking the trash icon button of the useControl hook.

Or how could I delete the selection made with the useControl hook without having to click on the trash icon button?

thanks

**Comment by Pessimistress** [comment]
> I have been trying to assign a reference inside the DrawControl component where I use the useControl hook, but I don't know-how

https://reactjs.org/docs/forwarding-refs.html

**Reply by raulrod16124** [reply]
Thanks for the suggestion, I've been trying different combinations of forwardRef but I'd say that this case is a bit different from the usual ref forwarding to a DOM element.

That is because the MapboxDraw plugin is presented as a JS object and the internal DOM is abstracted by react-map-gl.

The drawControl is based on the example at https://github.com/visgl/react-map-gl/blob/master/examples/draw-polygon/src/draw-control.ts

I've done a few experiments in trying to assign a value to the incoming refprop.

As well as an attempt to follow the pattern you use for MapRef but I've not managed to find the right way.

Any tips on how could I adapt the drawControl.tsx that I've published in the sandbox to pass the ref to the interal MapboxDraw plugin?
